clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerHeartbeatRequestManagerTest.java

```diff
@@ -602,6 +602,44 @@ public void testHeartbeatResponseOnErrorHandling(final Errors error, final boole
         }
     }
 
+    @Test
+    public void testUnsupportedVersion() {
+        mockErrorResponse(Errors.UNSUPPORTED_VERSION, Errors.UNSUPPORTED_VERSION.message());
+        ArgumentCaptor<ErrorEvent> errorEventArgumentCaptor = ArgumentCaptor.forClass(ErrorEvent.class);
+        verify(backgroundEventHandler).add(errorEventArgumentCaptor.capture());
+        ErrorEvent errorEvent = errorEventArgumentCaptor.getValue();
+
+        // UnsupportedApiVersion in HB response without any custom message. It's considered as new protocol not supported.
+        String hbNotSupportedMsg = "The cluster does not support the new consumer group protocol. Set group" +
+            ".protocol=classic on the consumer configs to revert to the classic protocol until the cluster is upgraded.";
+        assertInstanceOf(Errors.UNSUPPORTED_VERSION.exception().getClass(), errorEvent.error());
+        assertEquals(hbNotSupportedMsg, errorEvent.error().getMessage());
+        clearInvocations(backgroundEventHandler);
+
+        // UnsupportedApiVersion in HB response with custom message. Specific to required version not present, should
+        // keep the custom message.
+        String hbVersionNotSupportedMsg = "The cluster does not support resolution of SubscriptionPattern on version 0. " +
```

Comment:
Do we have a guard on the server side? I assumed that the `SubscriptionPattern` check would be within `throwIfConsumerGroupHeartbeatRequestIsInvalid`, but I was unable to locate it.

https://github.com/apache/kafka/blob/6fd951a9c0aa773060cd6bbf8a8b8c47ee9d2965/group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java#L1153

Reply:
I am sorry but I don’t understand what you mean. On the server, we basically handle the field if it is provided.

Reply:
ok, this is intended to verify that the client generates the Unsupported error with a custom msg (on build, so it really never makes it to the broker)